Featrue/residential neighborhood

AmmarAbudrhman opened the pull request.

Commit:
feat : add class for ResidentialNeighborhood

Commit:
feat: update the ReturnResidentialUnit to include         public int ResidentialNeighborhoodId { get; set; }
        public ResidentialNeighborhood ResidentialNeighborhood { get; set; } = null!;

Commit:
feat: add dto for the feature

Commit:
feat: add configuration for the feature

Commit:
add migration and interface for the feature

Commit:
add routing and dto and interface

Commit:
complete the feature

Dev7-Mubarak reviewed (changes requested):
Good job, do the following: 
 

1. Remove pagination and search functionality from ResidentialNeighborhood endpoints.
2. Introduce a new role: `ResidentialNeighborhoodManager`.
3. When creating a ResidentialNeighborhood, assigning a ResidentialNeighborhoodManager is required.
4. Restrict access to all ResidentialNeighborhood endpoints so that only users with the Admin role can access them.

Commit:
Remove pagination and search functionality from ResidentialNeighborhood endpoints.
Introduce a new role: ResidentialNeighborhoodManager.
When creating a ResidentialNeighborhood, assigning a ResidentialNeighborhoodManager is required.
Restrict access to all ResidentialNeighborhood endpoints so that only users with the Admin role can access them.

Dev7-Mubarak committed:
Add ResidentialNeighborhoodManager role migration

Added EF Core migration to seed "ResidentialNeighborhoodManager" role with a specific GUID. Updated seeded admin user credentials (ConcurrencyStamp, PasswordHash, SecurityStamp). Updated migration designer and ApplicationDbContextModelSnapshot accordingly.

Commit:
complete all the changes required
